cmd/admin-monitor.go

```diff
@@ -0,0 +1,202 @@
+	msg += "\n"
+
+	msg += fmt.Sprintf("%s        min        avg      max\n", console.Colorize("Monitor", "   CPU"))
+	for i := range s.CPULoad.Load {
+		msg += fmt.Sprintf("   current    %.2f%%      %.2f%%    %.2f%%\n", s.CPULoad.Load[i].Min, s.CPULoad.Load[i].Avg, s.CPULoad.Load[i].Max)
+		msg += fmt.Sprintf("   historic   %.2f%%      %.2f%%    %.2f%%\n", s.CPULoad.HistoricLoad[i].Min, s.CPULoad.HistoricLoad[i].Avg, s.CPULoad.HistoricLoad[i].Max)
+		msg += "\n"
+	}
+
+	// Mem section
+	msg += fmt.Sprintf("%s        usage\n", console.Colorize("Monitor", "   MEM"))
+	for i := range s.MemUsage.Usage {
+		msg += fmt.Sprintf("   current    %s\n", humanize.IBytes(s.MemUsage.Usage[i].Mem))
+		msg += fmt.Sprintf("   historic   %s\n", humanize.IBytes(s.MemUsage.HistoricUsage[i].Mem))
+		msg += "\n"
+	}
+
+	return msg
+}
+
+func mainAdminMonitor(ctx *cli.Context) error {
+	checkAdminMonitorSyntax(ctx)
+
+	// set the console colors
+	console.SetColor("Monitor", color.New(color.FgGreen, color.Bold))
+	console.SetColor("MonitorDegraded", color.New(color.FgYellow, color.Bold))
+	console.SetColor("MonitorFail", color.New(color.FgRed, color.Bold))
+
+	// Get the alias
+	args := ctx.Args()
+	aliasedURL := args.Get(0)
+
+	// Create a new minio admin client
+	client, err := newAdminClient(aliasedURL)
+	fatalIf(err, "Cannot get a configured admin connection.")
+
+	printOfflineMessage := func() {
+		printMsg(serverMonitorMessage{
+			Addr:    aliasedURL,
+			Service: "off",
+		})
+	}
+
+	processErr := func(e error) error {
+		switch e.(type) {
+		case *json.SyntaxError:
+			printOfflineMessage()
+			return e
+		case *url.Error:
+			printOfflineMessage()
+			return e
+		default:
+			// If the error is not nil and unrecognized, just print it and exit
+			fatalIf(probe.NewError(e), "Cannot get service status.")
+		}
+		return nil
+	}
+	// Fetch info of all CPU loads (all minio server instances)
+	cpuMonitors, e := client.ServerCPULoadInfo()
+	if err := processErr(e); err != nil {
+		// exit immediately if error encountered
+		return nil
+	}
+
+	memUsages, e := client.ServerMemUsageInfo()
+	if err := processErr(e); err != nil {
+		// exit immediately id error encountered
+		return nil
+	}
+
+	for i, cpuMonitor := range cpuMonitors {
+		memUsage := memUsages[i]
+		if cpuMonitor.Error != "" {
+			printMsg(serverMonitorMessage{
+				Service: "on",
+				Addr:    cpuMonitor.Addr,
+				Err:     cpuMonitor.Error,
+			})
+			continue
+		}
+		if memUsage.Error != "" {
+			printMsg(serverMonitorMessage{
+				Service: "on",
+				Addr:    memUsage.Addr,
+				Err:     memUsage.Error,
+			})
+			continue
+		}
+
+		printMsg(serverMonitorMessage{
+			Service:  "on",
+			Addr:     cpuMonitor.Addr,
+			CPULoad:  &cpuMonitor,
+			MemUsage: &memUsage,
+		})
+	}
+	return nil
+}
+
+// checkAdminMonitorSyntax - validate all the passed arguments
+func checkAdminMonitorSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) == 0 || len(ctx.Args()) > 2 {
+		exit := 1
```

Comment:
We have a global const for exit status to indicate failure, namely `globalErrorExitStatus`.

Reply:
sounds good